packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -81,116 +92,505 @@ const propTypes = {
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const momentDateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormatOrder = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateSeparator = intl.formatMessage({ id: 'Terra.datePicker.separator' });
+  const label = ariaLabel || intl.formatMessage({ id: 'Terra.datePicker.date' });
+  const placeholderValues = DateUtil.getDateInputValues(dateFormatOrder, placeholder, dateSeparator);
+  const theme = React.useContext(ThemeContext);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
     }
-  }
+  }, [shouldShowPicker, onClick]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, momentDateFormat)) {
+      const parseDate = DateUtil.getDateInputValues('YMD', dateValue, '-');
+      dateDispatch(parseDate);
+    } else if (dateValue === '') {
+      dateDispatch(dateState);
+    }
+  }, [momentDateFormat, dateValue, value]);
 
-    if (!attributes.readOnly && this.onCalendarButtonClick && this.props.onClick) {
-      this.onCalendarButtonClick(event, this.props.onClick);
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {number} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateOrder - Date variant based on the currently supported date
+   *                      orders: 'MDY', 'DMY', or 'YMD'.
+   */
+  const moveFocusOnChange = (inputValue, type, dateOrder) => {
+    if (dateOrder === 'MDY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === 'DMY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
```

Comment:
combine these two if conditions (here and above) similar to below if block of 'YMD'